twoopstracker/authentication/views.py

```diff
@@ -1,9 +1,30 @@
+from allauth.socialaccount import signals
+from allauth.socialaccount.models import SocialAccount
 from allauth.socialaccount.providers.oauth2.client import OAuth2Client
 from dj_rest_auth.registration.views import SocialLoginView
+from dj_rest_auth.views import UserDetailsView as BaseUserDetailsView
+from rest_framework.generics import RetrieveUpdateDestroyAPIView
 
 from .providers.googlesub.views import GoogleSubOAuth2Adapter
 
 
 class GoogleLogin(SocialLoginView):
     adapter_class = GoogleSubOAuth2Adapter
     client_class = OAuth2Client
+
+
+class UserDetailsView(BaseUserDetailsView, RetrieveUpdateDestroyAPIView):
+    def perform_destroy(self, instance):
+        social_account = SocialAccount.objects.filter(user=instance).first()
+        if social_account:
+            # NOTE(kilemensi): Since we're deleting it, skip any validation
+            #                  such as
+            # https://github.com/pennersr/django-allauth/blob/0.47.0/allauth/socialaccount/adapter.py
+            social_account.delete()
+            signals.social_account_removed.send(
```

Comment:
Given that we don't have any receiver for this signal, why are we sending it?

Reply:
That's how [**dj-rest-auth**](https://github.com/iMerica/dj-rest-auth) disconnects a [`SocialAccount`](https://github.com/iMerica/dj-rest-auth/blob/2d455333d5ae9910adb43de3488013ee2028f2eb/dj_rest_auth/registration/views.py#L216-L221).

Reply:
yes but for our case, unless we want to perform a certain action when a social account is deleted, I don't think we need to send out the signal.

Reply:
Why? For example Django sends signals whenever a model is saved, deleted, etc., and 99% of the the time, we don't use them. Why aren't we disabling them?

Unless there is a reason to break the expectation set by dj_rest_auth, P[rinciple of least astonishment](https://en.wikipedia.org/wiki/Principle_of_least_astonishment) should apply